lib/dao.js

```diff
@@ -1477,25 +1496,34 @@ DataAccessObject.update =
 DataAccessObject.updateAll = function (where, data, options, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
 
-  assert(arguments.length >= 2, 'At least two arguments are required');
-  if (options === undefined && cb === undefined) {
+  assert(arguments.length >= 1, 'At least one argument is required');
+
+  if (data === undefined && options === undefined && cb === undefined && arguments.length === 1) {
```

Comment:
My counter-example is that when a method forwards it's arguments, perhaps amending some of them along the way, then it may pass `undefined` value.

However, in this specific case, I am ok with the solution here.